WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationViewModel.swift

```diff
@@ -15,46 +30,46 @@ final class CommentModerationViewModel: ObservableObject {
         return .init(coreDataStack: ContextManager.shared)
     }()
 
-    // Temporary init argument
     init(
-        state: CommentModerationState,
         comment: Comment,
-        coordinator: CommentModerationCoordinator
+        coordinator: CommentModerationCoordinator,
+        notification: Notification?,
+        stateChanged: ((Result<CommentModerationState, CommentModerationError>) -> Void)?
     ) {
-        self.state = state
+        self.state = CommentModerationState(comment: comment)
         self.comment = comment
         self.coordinator = coordinator
+        self.notification = notification
+        self.stateChanged = stateChanged
     }
 
     func didChangeState(to state: CommentModerationState) {
```

Comment:
Nit: I find the name of this method misleading. Initially I thought the `state` property is already updated when this method is called. I think a more accurate name could be `didSelectState(_ state: CommentModerationState)`

Reply:
Resolved in [96c2bc](https://github.com/wordpress-mobile/WordPress-iOS/pull/23225/commits/396c2bc8b73f559bf05762260f960ccd5c693c41)